src/common/btree_class.c

```diff
@@ -228,6 +228,21 @@ nv_key_cmp(struct btr_instance *tins, struct btr_record *rec, d_iov_t *key)
 			key->iov_len));
 }
 
+static void
+nv_key_encode(struct btr_instance *tins, d_iov_t *key, daos_anchor_t *anchor)
+{
+	D_ASSERT(key->iov_len <= DAOS_ANCHOR_BUF_MAX);
+	memcpy(&anchor->da_buf[0], key->iov_buf, key->iov_len);
+}
+
+static void
+nv_key_decode(struct btr_instance *tins, d_iov_t *key, daos_anchor_t *anchor)
+{
+	key->iov_buf = &anchor->da_buf[0];
```

Comment:
I don't think this is correct.

See ktr_key_encode/decode in src/vos/vos_tree.c

You need to encode the length into the anchor.  Otherwise, the decoded key length here changes betwen encode and decode (after decode, it is always DAOS_ANCHOR_BUF_MAX

Reply:
ah, yes, you are right.